src/components/filter/filter.type.ts

```diff
@@ -29,9 +32,24 @@ export interface FilterEditionProps {
     broadcastChannel: BroadcastChannel;
     itemSelectionForCopy: ItemSelectionForCopy;
     setItemSelectionForCopy: (selection: ItemSelectionForCopy) => void;
-    getFilterById: (id: string) => Promise<{ [prop: string]: any }>;
+    getFilterById: (id: string) => Promise<{
+        [FieldConstants.EQUIPMENT_TYPE]: string;
+        [EXPERT_FILTER_QUERY]?: RuleGroupTypeExport;
+        [FILTER_EQUIPMENTS_ATTRIBUTES]?: any[];
+    }>;
     activeDirectory?: UUID;
     elementExists?: ElementExistsType;
     language?: string;
     description?: string;
 }
+
+export interface NewFilterType {
+    id: string | null;
+    type: string;
+    equipmentType: string;
+    rules?: RuleGroupTypeExport;
+    filterEquipmentsAttributes?: {
+        equipmentID: string;
+        distributionKey?: string;
+    }[];
```

Comment:
You can extract that 
```
type EquipmentsFilter = {
    equipmentID: string;
    distributionKey?: number;
};
```
and use it in the getFilterId above to replace the any. And I think distributionKey should be number, according to what I found in `explicitNamingFilterSchema`

Reply:
Done